core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -708,16 +709,36 @@ private[spark] class PythonBroadcast(@transient var path: String) extends Serial
       override def handleConnection(sock: Socket): Unit = {
         val env = SparkEnv.get
         val in = sock.getInputStream()
-        val dir = new File(Utils.getLocalDir(env.conf))
-        val file = File.createTempFile("broadcast", "", dir)
-        path = file.getAbsolutePath
-        val out = env.serializerManager.wrapForEncryption(new FileOutputStream(path))
+        val abspath = new File(path).getAbsolutePath
+        val out = env.serializerManager.wrapForEncryption(new FileOutputStream(abspath))
```

Comment:
just want to make sure I understand this part -- this change isn't necessary, right?  even in the old version, `path` gets updated here, so `setupDecryptionServer` would know where to read the data from.

that said, I do think your change makes more sense -- not sure why I didn't just use the supplied path in the first place.

Reply:
In the old version, we generated a random path with encryption turned off, so with encryption off it reads and writes from random path. When encryption related code was written we introduced a new "broadcast" path, the problem is when we tried to decrypt it on the driver side, it looks at the random path reference lying around and tries to decrypt from it but the actual data is in the new "broadcast" path location. So, by just passing the random reference, we make sure all the places are in sync with and without encryption

Reply:
yeah I see how it was wrong before.  I'm saying, after you add `setupDecryptionServer`, then that decryption server would still be reading from the value of `path` which gets updated here, since its the same object in the driver's JVM.

anyway, this isn't a big deal, I think its better with your change.

Reply:
ok I think we agree its good this way, (just to verify though I won't commit until you +1 it), but yes you are correct, now that we are using the decryption server which reads from the path in PythonBroadcast the path change isn't strictly necessary, but the value of self._path in broadcast.py doesn't match the path in PythonBroadcast so I think its better to have those match.

Reply:
yes +1.  Sorry didn't mean to get things stuck on this, just wanted to make sure I was actually following what was happening correctly.